lib/JsDOMEnvironment.js

```diff
@@ -0,0 +1,134 @@
+var Environment = require('./Environment'),
+    debug = require('debug')('JsDOMEnvironment'),
+    _ = require('lodash'),
+    jsdom = require('jsdom')
+    path = require('path'),
+    vow = require('vow');
+
+const defaultOptions = {
+    // Custom environment options
+    snapshot: false,
+    snapshotDir: 'snapshots',
+    proxy: null,
+    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12',
+};
+
+/**
+ * @param {object} options
+ * @constructor
+ */
+function JsDOMEnvironment(options) {
+    debug('Initializing...');
+    this._options = _.defaults(_.clone(options) || {}, defaultOptions);
+    this._proxy = this._options.proxy;
+    this._url = options.url;
+
+    if (!this._url) {
+        throw new Error('You must pass `url` to JsDOMEnvironment');
+    }
+    this._window = this._document = null;
+}
+
+JsDOMEnvironment.prototype = _.create(Environment.prototype, /**@lends JsDOMEnvironment*/{
+
+    /**
+     * Prepare environment
+     * @returns {Promise}
+     */
+    prepare: function() {
+        debug('Preparing...');
+        const deferred = vow.defer();
+
+        const params = {
+            url: this._url,
+            // scripts: ['https://cdnjs.cloudflare.com/ajax/libs/sizzle/2.3.3/sizzle.min.js'],
+            scripts: ['file:' + path.resolve(__dirname, '../vendor/sizzle.min.js')],
+            done: (error, window) => {
+                debug('Page is initialized in JsDom');
+                if (error) {
+                    return deferred.reject(error);
+                }
+                deferred.resolve(this._window = window);
+            }
+        };
+        this._setProxy(params)
+            ._setUserAgent(params);
```

Comment:
Do not sure these methods should be "private". Even if they are private, these should be written as:

```
this._setProxy(params);
this._setUserAgent(params);
```

But I believe we should start migration process to airbnb code style guide, and it claims [do not use dangling underscores](https://github.com/airbnb/javascript#naming--leading-underscore)